src/orbits/keplerian.jl

```diff
@@ -0,0 +1,532 @@
+        duration = nt.duration
+    elseif !isnothing(nt.incl)
+        !isnothing(nt.duration) && throw(ArgumentError(
+            "Only `incl`, `b`, or `duration` can be given"
+        ))
+        incl = nt.incl
+        sin_incl, cos_incl = sincos(incl)
+        b = cos_incl / dcosi_db
+        duration = nt.duration
+    elseif !isnothing(nt.duration)
+        duration = nt.duration
+        c = sin(π * duration / (incl_factor_inv) / period)
+        c_sq = c^2
+        ecc_sin_omega = ecc*sin_omega
+        aor = a_planet / R_star
+        b = √(
+            (aor^2 * c_sq - 1.0) /
+            (
+                c_sq * ecc_sin_omega^2 +
+                2.0*c_sq*ecc_sin_omega +
+                c_sq - ecc^4 + 2.0*ecc^2 - 1.0
+            )
+        ) * (1.0 - ecc^2)
+        cos_incl = dcosi_db * b
+        incl = acos(cos_incl)
+        sin_incl = sin(incl)
+    else
+        incl = 0.5*π
+        cos_incl = 0.0
+        sin_incl = 1.0
+        b = 0.0
+        duration = nt.duration
+    end
+
+    # Compute remaining system parameters
+    !(isnothing(nt.t_0) ⊻ isnothing(nt.t_p)) && throw(
+        ArgumentError("Either t₀ or tₚ must be specified")
+    )
+    if isnothing(nt.t_0)
+        t_p = nt.t_p
+        t_0 = t_p + M_0 / n
+    else
+        t_0 = nt.t_0
+        t_p = t_0 - M_0 / n
+    end
+
+    t_ref = t_p - t_0
+
+    # Normalize inputs
+    (
+        period, t_0, t_p, t_ref, duration,
+        a, a_planet, a_star, R_planet, R_star,
+        rho_planet, rho_star,
+        RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega,
+        incl, omega, Omega,
+        M_planet, M_star,
+    ) = normalize_inputs(
+        period, t_0, t_p, t_ref, duration,
+        a, a_planet, a_star, R_planet, R_star,
+        rho_planet, rho_star,
+        RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega,
+        incl, omega, Omega,
+        M_planet, M_star,
+        no_units,
+    )
+    return KeplerianOrbit(
+        period, t_0, t_p, t_ref, duration,
+        a, a_planet, a_star, R_planet, R_star,
+        rho_planet, rho_star,
+        RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega,
+        incl, omega, Omega,
+        n,
+        M_planet, M_star,
+    )
+end
+
+@kwcall KeplerianOrbit(
+    period=nothing, t_0=nothing, t_p=nothing, duration=nothing,
+    a=nothing, R_planet=nothing, R_star=nothing,
+    rho_star=nothing,
+    RpRs=nothing, aR_star=nothing, b=nothing, ecc=nothing, cos_omega=nothing, sin_omega=nothing,
+    incl=nothing, omega=nothing, Omega=nothing,
+    M_planet=nothing, M_star=nothing,
+)
+
+@kwalias KeplerianOrbit [
+    P => period,
+    t₀ => t_0,
+    tₚ => t_p,
+    τ => duration,
+    Rₚ => R_planet,
+    Rₛ => R_star,
+    ρₛ => rho_star,
+    aRₛ => aR_star,
+    cos_ω => cos_omega,
+    sin_ω => sin_omega,
+    ω => omega,
+    Ω => Omega,
+    Mₚ => M_planet,
+    Mₛ => M_star,
+]
```

Comment:
I think your alias/convention changes make a lot of sense! Just two quick follow-up question:
1) Did you mean to include the subscript versions in the suggested changes above as well, or do you want to stick with, e.g., `t0 => t_0` over `t₀ => t_0`
2) In terms of searchability,  I have been leaning towards using the longer version of symbols when possible. For example, using `ecc` over `e`. Since you mentioned using `r` internally in your other comment here https://github.com/JuliaAstro/Transits.jl/pull/23#discussion_r702188357, do you want to just switch to using the shortform aliases of the other variables for consistency  as well?

Reply:
Sounds good! For aesthetic purposes, I also updated the pretty printing accordingly using a mix of subscripted or standard notation where it made sense, but happy to switch it all over to one convention if you think that would be clearer for the user. edae1ee

Reply:
Actually I am having second thoughts about using subscripts in the printed output. I think it would probably be a pain if the user saw something like ` Mₛ: 1.0 M⊙` in the output for `orbit::KeplerianOrbit`, but then could only access that field through `orbit.M_star`